File: c#/EmployeeManagement/Models/PTEmployee.cs
using System;
using EmployeeManagement.Exceptions;
using InvalidDataException = EmployeeManagement.Exceptions.InvalidDataException;

namespace EmployeeManagement.Models
{
    // Base class for part-time employees who are paid by hour.
    public class PTEmployee : Employee
    {
        public double Rate { get; }
        public int Hours { get; }

        public PTEmployee(int id, string firstName, string lastName, DateOnly dateOfBirth, string position, Departments department, double rate, int hours)
            : base(id, firstName, lastName, dateOfBirth, position, department)
        {
            if (rate < 0)
            {
                throw new InvalidDataException("Rate cannot be negative.");
            }

            if (hours < 0)
            {
                throw new InvalidDataException("Hours cannot be negative.");
            }

            Rate = rate;
            Hours = hours;
        }

        // Calculates and prints pay for part-time work.
        public override void GetPaid()
        {
            var payment = Rate * Hours;
            Console.WriteLine($"{Position} {FirstName} {LastName} receives a part-time payment of {payment:C} for {Hours} hours at {Rate:C} per hour.");
        }

        // Includes bonus payment for part-time employees.
        public override void GetPaid(double bonus)
        {
            if (bonus < 0)
            {
                throw new PayrollMathException("Bonus cannot be negative.");
            }

            var payment = Rate * Hours + bonus;
            Console.WriteLine($"{Position} {FirstName} {LastName} receives a part-time payment of {payment:C} including bonus {bonus:C}.");
        }

        public override void ReportToManager()
        {
            Console.WriteLine($"{Position} {FirstName} {LastName} reports to the manager with a part-time employee schedule.");
        }

        public override string ToString()
        {
            return base.ToString() + $", Rate={Rate:C}, Hours={Hours}";
        }
    }
}

File: Java/PTEmployee.java
public class PTEmployee extends Employee {

    private double rate;
    private int hours;

    public PTEmployee(
        int id,
        String first_name,
        String last_name,
        String dob,
        String position,
        Departments department,
        double rate,
        int hours
    ) throws InvalidDataException {
        super(id, first_name, last_name, dob, position, department);
        setRate(rate);
        setHours(hours);
    }

    @Override
    public void getPaid() {
        try {
            if (rate < 0 || hours < 0) {
                throw new PayrollMathException("Rate and hours must be zero or greater.");
            }
            System.out.println("Part-time pay for " + getFirstName() + " " + getLastName() + ": " + (rate * hours));
        } catch (PayrollMathException exception) {
            System.out.println("Payroll error for employee " + getId() + ": " + exception.getMessage());
        }
    }

    @Override
    public void reportToManager() {
        System.out.println(getFirstName() + " " + getLastName() + " reports to Manager.");
    }

    public double getRate() {
        return rate;
    }

    public void setRate(double rate) throws InvalidDataException {
        if (rate < 0) {
            throw new InvalidDataException("Rate must be zero or greater.");
        }
        this.rate = rate;
    }

    public int getHours() {
        return hours;
    }

    public void setHours(int hours) throws InvalidDataException {
        if (hours < 0) {
            throw new InvalidDataException("Hours must be zero or greater.");
        }
        this.hours = hours;
    }
}
